packages/components/src/vue/vuex-modules/data/lexis.js

```diff
@@ -77,6 +150,26 @@ Lexis.store = (moduleInstance) => {
 
       hideCedictNotification (state) {
         state.cedictDisplayNotification = false
+      },
+
+      /**
+       * Sets treebank information in a Vuex store
+       *
+       * @param {object} state - A Vuex state object
+       * @param {TreebankDataItem} treebankDataItem - A treebank data item element
+       */
+      setTreebankInfo (state, treebankDataItem) {
+        state.hasTreebankData = treebankDataItem.hasTreebankData
+        state.treebankSrc = treebankDataItem.fullUrl
+      },
+
+      resetTreebankInfo (state) {
+        state.hasTreebankData = false
+        state.treebankSrc = null
```

Comment:
I think the next lookup will re-parse the page during creation of a `TreebankDataItem` object. If successful, it will cal `setTreebankInfo` that will set the `treebankSrc` URL. But that might not be fast enough to run queries against it (but we could add a `refreshUntilLoaded()` call before a query to accommodate for that). 

It might be more efficient not to reset a URL every time a lookup is started but rather use a separate variable that will disable treebank queries if a lookup without a context is used, and change a URL only if source changes. Probalby a lot depends on how the treebank app works internally.

Reply:
>It might be more efficient not to reset a URL every time a lookup is started but rather use a separate variable that will disable treebank queries if a lookup without a context is used, and change a URL only if source changes. Probalby a lot depends on how the treebank app works internally.

Yes, that's what I'm thinking. We need to do some more work here to track the state of the lookups vs the state of the document loaded in Arethusa.  I will try to describe this more clearly in a separate issue.